tests/reference/cpp-doconcurrentloop_01-4e9f274.stdout

```diff
@@ -28,11 +28,11 @@ void main0()
     Kokkos::View<float[10000]> c("c");
     int nsize;
     float scalar;
-    scalar = 10;
+    scalar = (float)(10);
```

Comment:
This is actually incorrect.  This is a bug in a test, it should give a compiler error message saying that you cannot assign integer to a variable that is declared as a float:
```python
    scalar: f32
    scalar = 10
```
And the fix is to use `scalar = 10.0`.

Reply:
> This is actually incorrect.

I think this should be allowed to assign an integer to float. Python also allows to do that:
```py
>>> d: float
>>> d = 2
>>> print(d)
2

```

Reply:
Fortran and C allows to do that. But not Python:
```
>>> d: float
>>> d = 2
>>> print(d)
2
>>> type(d)
<class 'int'>
```
And we should emulate what Python does. In python, `d` will be an integer. So that's why if we declare `d: float`, we need to give an exception in LPython.

Reply:
Oh, okay, that makes sense!